container-engine-lib/lib/backend_impls/docker/object_attributes_provider/docker_label_key/docker_label_key.go

```diff
@@ -20,25 +20,59 @@ type DockerLabelKey struct {
 }
 
 // NOTE: This is ONLY for areas where the label is declared statically!! Any sort of dynamic/runtime label creation
-// should use CreateNewDockerLabelKey
+// should use createNewDockerLabelKey
 func MustCreateNewDockerLabelKey(str string) *DockerLabelKey {
-	key, err := CreateNewDockerLabelKey(str)
+	key, err := createNewDockerLabelKey(str)
 	if err != nil {
 		panic(err)
 	}
 	return key
 }
 
-func CreateNewDockerLabelKey(str string) (*DockerLabelKey, error) {
-	if !dockerLabelKeyRegex.MatchString(str) {
-		return nil, stacktrace.NewError("Label key string '%v' doesn't match Docker label key regex '%v'", str, dockerLabelKeyRegexStr)
+// CreateNewDockerUserCustomLabelKey creates a custom uer Docker label with the Kurtosis custom user prefix
+func CreateNewDockerUserCustomLabelKey(str string) (*DockerLabelKey, error) {
+	if err := validateNotEmptyUserCustomLabelKey(str); err != nil {
+		return nil, stacktrace.NewError("Received an empty user custom label key")
 	}
-	if len(str) > maxLabelLength {
-		return nil, stacktrace.NewError("Label key string '%v' is longer than max label key length '%v'", str, maxLabelLength)
+	labelKeyStr := customUserLabelsKeyPrefixStr + str
+	return createNewDockerLabelKey(labelKeyStr)
+}
+
+func ValidateUserCustomLabelKey(str string) error {
+	if err := validateNotEmptyUserCustomLabelKey(str); err != nil {
+		return stacktrace.Propagate(err, "Received an empty user custom label key")
+	}
+	labelKeyStr := customUserLabelsKeyPrefixStr + str
+	if err := validate(labelKeyStr); err != nil {
+		return stacktrace.Propagate(err, "User custom label key '%s' is not valid", str)
+	}
+	return nil
+}
+
+func createNewDockerLabelKey(str string) (*DockerLabelKey, error) {
+	if err := validate(str); err != nil {
+		return nil, stacktrace.NewError("Label key string '%v' is not valid", str)
 	}
 	return &DockerLabelKey{value: str}, nil
 }
 
 func (key *DockerLabelKey) GetString() string {
 	return key.value
 }
+
+func validateNotEmptyUserCustomLabelKey(str string) error {
+	if str == "" || str == " " {
```

Comment:
what if its "    "; can we just strip the string and see if its empty?

Reply:
yup! that is better. I'll do it